doc/source/extended_documentation/calibration/rainforests_calibration/rainforests_calibration.rst

```diff
@@ -0,0 +1,173 @@
+taking an appropriate set of feature parameters (meteorological variables) as inputs.
+Details of this method can be found in `Hewson & Pillosu, 2021`_. 
+
+****************************
+The RainForests method
+****************************
+
+RainForests is an adaptation of the ECPoint method that seeks to use machine learning based
+tree models to replace the manually trained tree model of ECPoint. Here we use gradient-boosted
+decision tree (GBDT) ensemble models.
+
+The underlying principle of the calibration methodology is essentially the same, namely using
+a set of feature variables to map onto error distributions which can be applied to the
+forecast values to produce calibrated forecast values. However, the way in which these
+distributions are constructed differs somewhat within the RainForests framework.
+
+Our approach is to use a series of GBDT models, taking the feature variables as inputs, to
+produce exceedance probability values for representative error thresholds. Collectively
+these represent the error distributions as a cumulative distribution function.
+
+Each CDF is then mapped onto a series of equispaced percentile values to yield a series of
+representative error values which can be applied to the grid-scale forecast value to produce
+the series of point-scale (calibrated) values.
+
+Calibration of the ensemble forecast proceeds by determining the underlying distribution on
+a per realization basis and applying this to each forecast value to produce a series of
+distinct calibrated forecast values for each realization. Collectively these values form the
+so-called super-ensemble, which we subsequently sample to produce the calibrated forecast
+ensemble output.
+
+Another point of difference between RainForests and ECPoint is in the choice of error value
+underlying the error distribution. For RainForests we have chosen to use additive error in
+place of a multiplicative error (forecast error ratio in ECPoint) to allow us to calibrate
+input forecast values of zero-rainfall.
+
+===========================
+GBDT vs manually trained DT
+===========================
+
+The choice of using GBDT models in place of the manually trained DT of ECPoint comes with
+some advantages, but at the expense of some trade-offs:
+
+**Advantages:**
+
+* GBDT is an ensemble of many trees rather than a single tree. This means outputs are
+  near-continuous relative to the inputs.
+* Trees are built algorithmically, not manually, with each branch split automatically
+  chosen to be optimal relative to some loss function. In principle this gives better
+  accuracy, and makes it easier to retrain on new data.
+
+**trade-offs:**
+
+* By using an ensemble of trees, the intuitive connection between weather type and feature
+  variables is lost.
+* Using a series of decision tree ensembles in place of a single decision tree increases the
+  computational demand significantly.
```

Comment:
Another trade-off is that some effort is required to select a good set of model hyper-parameters that neither under-fit nor overfit. (This is also somewhat true for the original ECPoint approach, i.e. one wouldn't want too many or too few leaves in the tree, but the problem is more transparent there because it's easy to see exactly how the data is being partitioned into leaves).

Reply:
I agree, however I think that there is also an advantage here in that this is something that in principle should only be required once. In the case of ECPoint, the problem is more transparent in that it amounts to ensuring leaves have sufficient data, however this is something that must be checked each time the trees are reconstructed.

How about this:
```
* Some initial effort is required to select a good set of model hyper-parameters that neither under- or over-fit. This process is more challenging and less transparent than ECPoint, however is required only once rather than each time the decision tree(s) are constructed.

Reply:
Yep, I think this captures it well.

Reply:
Great! I'll add this point in.